src/wallet/scriptpubkeyman.h

```diff
@@ -242,6 +242,9 @@ class ScriptPubKeyMan
 
     virtual uint256 GetID() const { return uint256(); }
 
+    /** Returns a vector of all the scriptPubKeys that this ScriptPubKeyMan watches */
+    virtual const std::unordered_set<CScript, SaltedSipHasher> GetScriptPubKeys() const { return {}; };
```

Comment:
220fe61f70f6093a695fd8b6e38e928b63c2b6ad: I assume the goal here is to get faster lookups and random access deletions? E.g. in the `erase()` call in the next commit. 

Afaik there's no guarantee in the standard library that `std::unordered_set` will put elements in the same order between runs. So if deterministic behaviour is the goal of this change, it might be better to just stick to `std::vector` but call `sort()` on it before returning. But that's slower.

Code comment nit: not a "vector"

Suggest renaming commit to: "scriptpubkeyman: Implement GetScriptPubKeys for Legacy" (the current title suggests it's an entirely new method)

Reply:
We don't need deterministic behavior. That's why I elected to use `std::unordered_set` and `SaltedSipHasher` (which is initialized with random salts).

Reply:
Fixed the comment and commit message.